### ISSUE TITLE
Vault checkout integration should be using a REST Api instead of rendering via phtml.

### ISSUE THREAD

**Issue by paales** [magento/payment-improvements#15]
Currently the payment form seems to be renderer server side, but this is causing issues with outr custom checkout. Having a REST api available to fetch the vault entries for a customer would be great.

**Comment by joni-jones** [comment]
Actually, the current Vault implementation renders all templates via js (uses knockoutjs) - `Magento/Vault/view/frontend/web/template/pyament/form.html`. Only `*.phtml` rendered during AJAX calls but this justly only for Admin panel because Sales in Admin panel is legacy and doesn't provide needed API.

**Comment by paales** [comment]
You are right, didn't look right. With out custom checkout we now just used the window.checkoutConfig.payment.vault path to get the payment methods, but that feels wrong. For now I don't have any pending issues.

**Comment by vkublytskyi** [comment]
Moved to discussions as we need to agree what should be implemnted
